deploy-platform/src/lib/workflows/Workflows.ts

```diff
@@ -44,6 +72,35 @@ export const Workflows: WorkflowsConstructor = class Workflows implements Workfl
     this._logger.info(`Workflow ${options.workflowId} dispatched`);
   }
 
+  public async findWorkflowToDispatch(workflowFileName: string): Promise<number> {
+    this._logger.info(
+      `Searching workflow ${workflowFileName} in repository workflows list to dispatch`
+    );
+    const workflows = await this._githubClient.getWorkflows();
+    this._logger.debug(`Workflows found: ${JSON.stringify(workflows.data.workflows)}`);
+    const foundWorkflowId = this._findWorkflowIdInWorkflowsResponse(
+      workflows.data.workflows,
+      workflowFileName
+    );
+    return foundWorkflowId;
+  }
+
+  private _findWorkflowIdInWorkflowsResponse(
+    workflows: GetWorkflowsResponse["data"]["workflows"],
+    workflowFileName: string
+  ): number {
+    const foundWorkflow =
+      findWorkflowWithPathEqualToLowercaseName(workflows, workflowFileName) ||
+      findWorkflowWithPathEqualsToLowercaseNameReplacingDashes(workflows, workflowFileName) ||
+      findWorkflowByName(workflows, workflowFileName);
+    if (!foundWorkflow) {
+      throw new Error(`Workflow ${workflowFileName} not found`);
+    }
+    this._logger.info(`Found workflow ${foundWorkflow.name} matching with ${workflowFileName}`);
```

Comment:
Should we display the file name instead of the name?
```suggestion
    this._logger.info(`Found workflow with fileName "${foundWorkflow.path.split("/").pop()}" and name "${foundWorkflow.name}" matching with "${workflowFileName}"`);
```